src/mnode/mnode-pastel.cpp

```diff
@@ -854,24 +921,31 @@ bool CArtSellTicket::IsValid(bool preReg, int depth) const
             existingTicket.m_txid == m_txid) // if this ticket is already in the DB
             ticketFound = true;
     }
-    
-    //1. check PastelID in this ticket matches PastelID in the referred ticket (Activation or Trade)
-    //2. Verify the art is not already sold
-    auto existingTradeTickets = CArtTradeTicket::FindAllTicketByArtTnxID(artTnxId);
-    auto soldCopies = existingTradeTickets.size();
-    auto existingSellTickets = CArtSellTicket::FindAllTicketByArtTnxID(artTnxId);
-    auto sellTicketsNumber = existingSellTickets.size();
-    auto totalCopies = 0;
+
+    // Check PastelID in this ticket matches PastelID in the referred ticket (Activation or Trade)
+    auto totalCopies{0};
+    // Verify the NFT is not already sold or gifted
+    const auto verifyAvailableCopies = [this](const std::string& strTicket, unsigned short totalCopies) {
```

Comment:
const size_t totalCopies, will then be compared with soldCopies which is vector::size() -> size_t

Reply:
ok

Reply:
fixed